src/codegen.ts

```diff
@@ -0,0 +1,43 @@
+import { isObject, isString } from 'lodash';
+import { CodeGenOptions, ProvidedCodeGenOptions, makeOptions } from './options/options';
+import { getViewForSwagger2 } from './getViewForSwagger2';
+import { transformToCodeWithMustache } from './transform/transformToCodeWithMustache';
+import { enhanceCode } from './enhance';
+
+function getCode(opts: CodeGenOptions): string {
+    verifyThatWeAreGeneratingForSwagger2(opts);
+
+    const data = getViewForSwagger2(opts);
+    return transformToCodeWithMustache(data, opts.template, opts.mustache);
+};
+
+
+export const CodeGen = {
+    transformToViewData: getViewForSwagger2,
+    transformToCodeWithMustache,
+    getTypescriptCode: function(opts: ProvidedCodeGenOptions){
+        const options = makeOptions(opts);
+
+        return enhanceCode(getCode(options), options, 'typescript');
+    },
+    getCustomCode: function(opts: ProvidedCodeGenOptions){
+        verifyThatWeHaveRequiredTemplatesForCustomGenerationTarget(opts);
+        
+        const options = makeOptions(opts);
+
+        return enhanceCode(getCode(options), options, 'custom');
+    }
+};
+
+function verifyThatWeHaveRequiredTemplatesForCustomGenerationTarget(opts: ProvidedCodeGenOptions) {
+    // TODO: Why do we not check for the existence of the type template?
```

Comment:
probably it's worth considering to switch to es6 templates, imho it should eliminate a lot of complexity and simplify customisation

Reply:
Thats probably a pretty good idea, I assume you mean we can get rid of Mustache all together? :)